udata/core/dataset/api.py

```diff
@@ -97,7 +97,12 @@ def __init__(self):
     @staticmethod
     def parse_filters(datasets, args):
         if args.get('q'):
-            datasets = datasets.search_text(args['q'])
+            # Following code splits the 'q' argument by spaces to surround
+            # every word in it with quotes before rebuild it.
+            # This allows the search_text method to tokenise with an AND
+            # between tokens whereas an OR is used without it.
+            phrase_query = ' '.join([f'"{elem}"' for elem in args['q'].split(' ')])
+            datasets = datasets.search_text(phrase_query, language=current_app.config['DEFAULT_LANGUAGE'])
```

Comment:
It might be the combination of:
`phrase_query = ' '.join([f'"{elem}"' for elem in args['q'].split(' ')])`
and the stemming of the selected language. Removing one or another solves the problem.

Reply:
btw, we could brainstorm on the fact that building the phrase query to be an AND query explicitly is weird wiith the use of stop words as it does not seem to work.
`la base sirene` ans `base sirene` does not return the same number of datasets, despite the fact that it should be the same query once the query is stripped from  the stop word `la` .